File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    int s[] = {0,0,0};
    int a1[] = {1, 2, 3, 4, 5};
    int a2[] = {2, 1, 2, 3, 2, 4, 2, 5};
    int a3[] = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};

    for(int i=0;i<answers.size();i++){

        if(answers[i] == a1[i%5]){s[0]+=1;}
        if(answers[i] == a2[i%8]){s[1]+=1;}
        if(answers[i] == a3[i%10]){s[2]+=1;}
    }

    int max_i=0;
    for(int i=0;i<3;i++){
        if(s[i]>s[max_i]){max_i = i;}
    }
    answer.push_back(max_i+1);

    for(int i=max_i+1;i<3;i++){
        if(s[i]==s[max_i]){answer.push_back(i+1);}
    }

    return answer;
}

File: 프로그래머스/1/42840.모의고사-Python3/모의고사.py
def solution(answers):
    answer = []
    supo = [[1,2,3,4,5],
        [2,1,2,3,2,4,2,5],
        [3,3,1,1,2,2,4,4,5,5]]
    counts = [[0, 1], [0, 2], [0, 3]]
    for i in range(len(answers)):
        for j in range(3):
            if answers[i] == supo[j][i % len(supo[j])]:
                counts[j][0] += 1
    
    # counts.sort()
    counts.sort(key=lambda x: x[0], reverse=True)
    answer.append(counts[0][1])
    if counts[0][0] == counts[1][0]:
        answer.append(counts[1][1])
        if  counts[1][0] == counts[2][0]:
            answer.append(counts[2][1])

    return sorted(answer)